### PR TITLE
Add tuning script!

### PR DESCRIPTION
**Pull request by EthanMarx**
1. Add tuning script built entirely on [lightray](https://github.com/ethanmarx/lightray) that should work for both the `Similarity`  and `Flow` `LightningCLI` classes. If a ray cluster is active, can pass the ip to the `address` variable. Otherwise, will tune locally

2. Adds `amplfi_init.py` script for easily initializing fresh run directories with config files for training or tuning.
3. Adds `omegaconf` parser mode in `AmplfiBaseCLI` `__init__` so the cli is compatible with tuning script

### CODE REVIEW

**Comment by alecgunny** [comment]
@EthanMarx  Just chiming in to say I hadn't seen your `lightray` library and this whole thing is extremely sick

**Comment by EthanMarx** [comment]
@alecgunny Thank you thank you finally got around to generalizing things and turned out nicely!

**Comment by EthanMarx** [comment]
@deepchatterjeeligo Adds tuning script, and helper script (similar to what we have with aframe) for making initialization of run directories simpler. Will add some documentation on this.

File: projects/train/train/tune/tune.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    )
+
+    args = parser.parse_args()
+    init_logging(args.pop("verbose"))
+    args = parser.instantiate_classes(args)
+    train_config = args.pop("train_config")
+    args.pop("config")
+    train_args = ["--config", train_config]
+    # config = parse_args(args.cli_cls, train_args)
+    # print(config)
```
Comment (deepchatterjeeligo) [inline projects/train/train/tune/tune.py:47]: remove

File: scripts/amplfi_init.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    content = f"""
+    #!/bin/bash
+    # Export environment variables
+    export AMPLFI_DATADIR={base}
+    export AMPLFI_OUTDIR={base}/training/
+    export AMPLFI_CONDORDIR={path}/condor
+
+    # launch the data generation pipeline;
+    # note that if you've made local code changes not in the containers
+    # you'll need to add the --dev flag!
```
Comment (deepchatterjeeligo) [inline scripts/amplfi_init.py:107]: How is this used? `$ amplfi_init --dev`? Does the parser have this argument?

File: projects/train/train/cli/base.py [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 
 class AmplfiBaseCLI(LightningCLI):
+    def __init__(self, *args, **kwargs):
+        # hack into init to hardcode
+        # parser_mode to omegaconf for all subclasses
+        kwargs["parser_kwargs"] = {"parser_mode": "omegaconf"}
```
Comment (deepchatterjeeligo) [inline projects/train/train/cli/base.py:8]: What feature was not being supported because of omegaconf?

File: scripts/amplfi_init.py (hunk 2 above)
Comment (EthanMarx) [reply scripts/amplfi_init.py:107]: This refers the `--dev` flag for the `law` tasks. Not as relevant here where we only use law for querying data.

File: projects/train/train/cli/base.py (hunk 3 above)
Comment (EthanMarx) [reply projects/train/train/cli/base.py:8]: `omegaconf` adds parsing of environment variables (like `AMPLFI_OUTDIR`)

File: projects/train/train/tune/tune.py (hunk 1 above)
Comment (deepchatterjeeligo) [reply projects/train/train/tune/tune.py:47]:
```suggestion
```

File: scripts/amplfi_init.py (hunk 2 above)
Comment (deepchatterjeeligo) [reply scripts/amplfi_init.py:107]: ok let's remove the lines then to avoid confusion

**Comment by EthanMarx** [comment]
@deepchatterjeeligo should be good to go

**Pull request merged**